content/guides/end-to-end-testing/auth0-authentication.md

```diff
@@ -111,10 +117,146 @@ require('dotenv').config()
+
+Cypress.Commands.add('loginToAuth0', (username: string, password: string) => {
+  const log = Cypress.log({
+    displayName: 'AUTH0 LOGIN',
+    message: [`🔐 Authenticating | ${username}`],
+    // @ts-ignore
+    autoEnd: false,
+  })
+  log.snapshot('before')
+
+  // App landing page redirects to Auth0.
+  cy.visit('/')
+
+  // Login on Auth0 (we'll add caching later).
+  cy.origin(
+    Cypress.env('auth0_domain'),
+    { username, password },
+    ({ username, password }) => {
+      cy.get('input#username').type(username)
+      cy.get('input#password').type(password)
+      cy.contains('button[value=default]', 'Continue').click()
+    }
+  )
+
+  // Ensure Auth0 has redirected us back to the RWA.
+  cy.url().should('equal', 'http://localhost:3000/')
+
+  log.snapshot('after')
+  log.end()
+})
+```
+
+Now, we can use our `loginToAuth0` command in the test. Below is our test to
+login as a user via Auth0 and run a basic sanity check.
+
+<Alert type="success">
+
+The
+[runnable version of this test](https://github.com/cypress-io/cypress-realworld-app/blob/develop/cypress/tests/ui-auth-providers/auth0.spec.ts)
+is in the
+[Cypress Real World App](https://github.com/cypress-io/cypress-realworld-app).
+
+</Alert>
+
+```js
+describe('Auth0', function () {
+  beforeEach(function () {
+    cy.task('db:seed')
+    cy.intercept('POST', '/graphql').as('createBankAccount')
+    cy.loginToAuth0(
+      Cypress.env('auth0_username'),
+      Cypress.env('auth0_password')
+    )
+    cy.visit('/')
+  })
+
+  it('shows onboarding', function () {
+    cy.contains('Get Started').should('be.visible')
+  })
+})
+```
+
+Lastly, we can refactor our login command to take advantage of
+[`cy.session()`](/api/commands/session) to store our logged in user so we don't
+have to reauthenticate before every test.
+
+```js
+Cypress.Commands.add('loginToAuth0', (username: string, password: string) => {
+  const log = Cypress.log({
+    displayName: 'AUTH0 LOGIN',
+    message: [`🔐 Authenticating | ${username}`],
+    // @ts-ignore
+    autoEnd: false,
+  })
+  log.snapshot('before')
+
+  const args = { username, password }
+  cy.session(
+    args,
+    () => {
+      // App landing page redirects to Auth0.
+      cy.visit('/')
+
+      // Login on Auth0.
+      cy.origin(
+        Cypress.env('auth0_domain'),
+        { args },
+        ({ username, password }) => {
+          cy.get('input#username').type(username)
+          cy.get('input#password').type(password)
+          cy.contains('button[value=default]', 'Continue').click()
+        }
+      )
+
+      // Ensure Auth0 has redirected us back to the RWA.
+      cy.url().should('equal', 'http://localhost:3000/')
+    },
+    {
+      validate: () => {
+        // Validate presence of access token in localStorage.
+        cy.wrap(localStorage)
+          .invoke('getItem', 'authAccessToken')
+          .should('exist')
+      },
+    }
+  )
+
+  log.snapshot('after')
+  log.end()
+})
+```
```

Comment:
can we add something to the [Adapting the back end](https://github.com/cypress-io/cypress-documentation/pull/4895/files#diff-cb9f75c3460abdeff33c04572f09eb12e7c4ac69255ed0effeebcb12a2e1937bR407) section thats a disclaimer like:

```
<Alert type="info">

<strong class="alert-header">Programmatic Login</strong>

Unlike programmatic login, authenticating with
[`cy.origin()`](/api/commands/origin) does not require adapting the a backend to
work. This step is only needed if implementing programmatic login.

</Alert>
```

Reply:
Great idea